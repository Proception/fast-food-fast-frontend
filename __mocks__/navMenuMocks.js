import { setToken } from '../src/actions/authActions';

const stateGuest = {
    data: [],
    auth: {
        isAuthenticated: false,
    }
  };
  const stateUser = {
    data: [],
    auth: {
        isAuthenticated: true,
        user: {
          email: "omasan.esimaje@gmail.com",
          // full_name: "benedict"
      }
    },
    setToken: setToken,
    order: {
      cart: [1,2]
    }
  };
  const stateUserEmptyCart = {
    data: [],
    auth: {
        isAuthenticated: true,
        user: {
          email: "omasan.esimaje@gmail.com",
          // full_name: "benedict"
          role_id: 200
      }
    },
    setToken: setToken,
    order: {
      cart: []
    }
  };
  const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiZnVsbF9uYW1lIjoiQmVuZWRpY3QiLCJyb2xlX2lkIjoyMDAsImlhdCI6MTU1MTg3NzMyNSwiZXhwIjoxNTUxOTYzNzI1fQ.DwFFvZskGl9JeBnLD1CCnBSpYA1i0KgK1btUY1eLc9c'
  export { stateGuest, stateUser, token, stateUserEmptyCart };