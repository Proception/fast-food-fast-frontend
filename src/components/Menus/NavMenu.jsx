import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';

export class NavMenu extends Component {

  isAuthMenu() {
    return (

      <ul className="nav">
        <li><NavLink to="/" >Home</NavLink></li>
        <li><NavLink to="/menu" >Menu</NavLink></li>
        <li><NavLink to="/login" >Login</NavLink></li>
        <li><NavLink to="/signup" >Register</NavLink></li>
        <li><NavLink to="/checkout" >Checkout</NavLink></li>
        <li><NavLink to="#" >Settings</NavLink>
          <ul className="dropdown">
            <li><NavLink to="/orders" >View Pending Orders</NavLink></li>
            <li><NavLink to="/orderhistory" >View Order History</NavLink></li>
            <li><NavLink to="/search" >Search Food Items</NavLink></li>
            <li><NavLink to="/addfooditems" >Add Food Items</NavLink></li>
          </ul>
        </li>
      </ul>
    )
  }

  guestMenu() {
    return (
      <ul className="nav">
        <li><NavLink to="/" >Home</NavLink></li>
        <li><NavLink to="/menu" >Menu</NavLink></li>
        <li><NavLink to="/login" >Login</NavLink></li>
        <li><NavLink to="/signup" >Register</NavLink></li>
      </ul>
    )
  }

  render() {
    const { isAuthenticated } = this.props.auth;
    return (
      <header>
        <div className="container">
          <NavLink to="/" ><div id="brand-logo"><h1> <span className="">Fast</span> Food Fast</h1>
          </div></NavLink>
          <nav>
            {
              isAuthenticated ? this.isAuthMenu() : this.guestMenu()
            }
          </nav>
        </div>
      </header>
    )
  }
}

const mapStateToProps = state => ({
  auth: state.auth,
});

export default connect(mapStateToProps)(NavMenu);