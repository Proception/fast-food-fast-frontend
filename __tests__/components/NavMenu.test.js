import React from 'react';
import { NavMenu } from '../../src/components/Menus/NavMenu.jsx';
import { shallow } from 'enzyme';
import { stateGuest, stateUser } from '../../__mocks__/navMenuMocks.js';

describe('NavMenu Component', () => {
  it('should return guest menu', () => {
    const wrapper = shallow(<NavMenu { ...stateGuest }/>);
    expect(wrapper).toMatchSnapshot();
  });
  it('should return guest menu', () => {
    const wrapper = shallow(<NavMenu { ...stateUser }/>);
    expect(wrapper).toMatchSnapshot();
  });
});
