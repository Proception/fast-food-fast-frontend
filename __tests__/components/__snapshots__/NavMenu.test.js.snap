// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu Component should return guest menu 1`] = `ShallowWrapper {}`;

exports[`NavMenu Component should return guest menu 2`] = `ShallowWrapper {}`;
